tests/test.rs

```diff
@@ -1,51 +1,84 @@
 use os_pipe::pipe;
+use serial_test::serial;
 use std::{
     io::Read,
     process::{Command, ExitStatus},
-    sync::Mutex,
 };
-use structopt::lazy_static::lazy_static;
 
 struct RunResult {
     exit_status: ExitStatus,
     output: String,
 }
 
-lazy_static! {
-    /// rust will try to run the tests in parallel by default, and `insta` doesn't like the
-    /// usual way of disabling this via `--test-threads=1`, so we're using this
-    /// mutex to make sure we're not re-flashing until the last run is finished
-    static ref ONE_RUN_AT_A_TIME: Mutex<i32> = Mutex::new(0i32);
-}
-
 /// run probe-run with `args` and truncate the "Finished .. in .." and "Running `...`" flashing output
 /// NOTE: this currently only capures `stdin`, so any `log::` ed output, like flashing
-fn run_and_truncate(args: &str) -> RunResult {
-    let _guard = ONE_RUN_AT_A_TIME.lock().unwrap();
+fn run(args: &str) -> RunResult {
+    let (mut reader, mut handle) = run_command(args);
+
+    // retrieve output and clean up
+    let mut probe_run_output = String::new();
+    reader.read_to_string(&mut probe_run_output).unwrap();
+    let exit_status = handle.wait().unwrap();
+
+    // remove the lines printed during flashing, as they contain timing info that's not always the same
+    let output = truncate_output(probe_run_output);
+
+    RunResult {
+        exit_status,
+        output,
+    }
+}
+
+#[cfg(target_family = "unix")]
```

Comment:
from what I understand so far, simulating a ctrl+c on windows is a bit finicky (you'd probably use the [winapi](https://docs.rs/winapi/0.3.9/winapi/um/wincon/fn.GenerateConsoleCtrlEvent.html) crate which is a bit underdocumented?) and I don't have a windows machine at hand to play with this so I opted for this makeshift solution